apiserver/vshn/postgres/multiwatch.go

```diff
@@ -0,0 +1,41 @@
+package postgres
+
+import "k8s.io/apimachinery/pkg/watch"
+
+// MultiWatch is wrapper of multiple source watches which implements the same methods as a normal watch.Watch
+var _ watch.Interface = &MultiWatcher{}
+
+type MultiWatcher struct {
+	watchers []watch.Interface
+}
+
+// NewEmptyMultiWatch creates an empty watch
+func NewEmptyMultiWatch() *MultiWatcher {
+	return &MultiWatcher{}
+}
+
+// AddWatcher adds a watch to this MultiWatcher
+func (m *MultiWatcher) AddWatcher(w watch.Interface) {
+	m.watchers = append(m.watchers, w)
+}
+
+// Stop stops all watches of this MultiWatch
+func (m *MultiWatcher) Stop() {
+	for _, watcher := range m.watchers {
+		watcher.Stop()
+	}
+}
```

Comment:
We should also close the channel of the multiWatcher

Reply:
ah good catch, absolutely.